pkg/envfuncs/resource_funcs.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package envfuncs
+
+import (
+	"context"
+
+	"sigs.k8s.io/e2e-framework/klient/decoder"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+// SetupCRDs is provided as a helper env.Func handler that can be used to setup the CRDs that are required
+// to process your controller code for testing. This env.Func helper also provides a helper function that can be
+// used to programatically filter selective resources under question as well all support injecting resources.CreateOption
+func SetupCRDs(crdPath, pattern string, opts []resources.CreateOption, f decoder.PreHandlerFuncHook, decodeOpts ...decoder.DecodeOption) env.Func {
```

Comment:
I would make this function simpler, like:

```go
func SetupCRDs(crdPath, pattern string, opts []resources.CreateOption) env.Func { ... }
```
If someone needs more control, they can always call the `ApplyUsingManifestDir` which accepts more parameters.

Reply:
I will cleanup the `f decoder.PreHandlerFuncHook` as I am cleaning it up entirely. The `decodeOpts` is still helpful to be able to use right ?